File: app/schemas/duty.py
from typing import Optional
from pydantic import BaseModel

class DutyAssignRequest(BaseModel):
    date: str
    student_ids: list[int]

class DutyStudentRow(BaseModel):
    id: int
    name: str
    tg_id: int
    date: Optional[str] = None
    is_absent_now: bool

class DutiesResponse(BaseModel):
    duties: list[DutyStudentRow]

File: frontend/src/types/duty.ts
export interface DutyStudent {
  id: number;
  name: string;
  tg_id: number;
  date: string | null;
  is_absent_now: boolean;
}

export interface DutiesResponse {
  duties: DutyStudent[];
}

export interface DutyAssignRequest {
  date: string;
  student_ids: number[];
}
